### PR TITLE
fix(slo_controller): better events and statuses handling

### PR DESCRIPTION
**Pull request by fourstepper**

### CODE REVIEW

File: internal/controller/openslo/slo_controller.go [hunk 1]
```diff
@@ -95,7 +102,7 @@ func (r *SLOReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		err = utils.UpdateStatus(ctx, slo, r.Client, "Ready", metav1.ConditionFalse, "SLI Object not found")
 		if err != nil {
 			log.Error(err, "Failed to update SLO status")
-			r.Recorder.Event(slo, "Error", "SLIObjectNotFound", "SLI Object not found")
+			r.Recorder.Event(slo, "Warning", "SLIObjectNotFound", "SLI Object not found")
```
[the remark this answers is not in the document]
Comment (fourstepper) [reply internal/controller/openslo/slo_controller.go:105]: But like...

https://github.com/kubernetes/api/blob/6c11c9e4685cc62e4ddc8d4aaa824c46150c9148/core/v1/types.go#L6018-L6024

File: internal/controller/openslo/slo_controller.go [hunk 2]
```diff
@@ -161,7 +168,7 @@ func (r *SLOReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 		}
 
 		if err = r.Create(ctx, mimirRule); err != nil {
-			r.Recorder.Event(slo, "Error", "FailedToCreateMimirRule", "Failed to create Mimir Rule")
+			r.Recorder.Event(slo, "Warning", "FailedToCreateMimirRule", "Failed to create Mimir Rule")
```
[the remark this answers is not in the document]
Comment (fourstepper) [reply internal/controller/openslo/slo_controller.go:171]: https://github.com/oskoperator/osko/pull/74#discussion_r1518545885

File: internal/controller/openslo/slo_controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hy3n4) [reply internal/controller/openslo/slo_controller.go:105]: 🤔 I guess that makes sense then.

**Comment by fourstepper** [comment]
Should be merge-able now

File: internal/controller/openslo/slo_controller.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func (r *SLOReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 				log.Error(err, "Failed to update SLO ready status")
 				return ctrl.Result{}, err
 			}
-			return ctrl.Result{}, nil
+			return ctrl.Result{RequeueAfter: time.Second * 5}, nil
```
Comment (Hy3n4) [inline internal/controller/openslo/slo_controller.go:73]: Why do we need to do this?

Comment (fourstepper) [reply internal/controller/openslo/slo_controller.go:73]: That comes from this previous PR/commit https://github.com/oskoperator/osko/commit/13deb5f8f4717fddbd2957cfcf20868546e89a66#diff-843e6e74bf74fb2516f330ec7542781819a468e33325a9e82465e0524ad980d5R66

Comment (fourstepper) [reply internal/controller/openslo/slo_controller.go:73]: ... and solves the merge conflict that was here before :) check the diff of this branch vs master, this isn't actually changing between those two

Comment (Hy3n4) [reply internal/controller/openslo/slo_controller.go:73]: Hmm, I would say that this must have been one of my brain farts then :D

**Review by Hy3n4 — APPROVED** [review]
LGTM

**Pull request merged**